### PR TITLE
Final Group Capstone - Mentor Me Now - Front-end

### LINKED ISSUE #24: [1pt]-Submit project
- [ ] Submit the project for an external review from the dashboard

### PR DESCRIPTION
**Pull request by Ayokunnumi1**
Screenshot from 2024-02-14 19-48-39

This pull request aims to enhance the user interface of the Mentor Me Now application by implementing a responsive design and adding new features.

1. Navigation Panel Updates:

Addition of route links to key sections such as "Reserve Mentor", "Reservations", "Add Mentor", "Add Reservation" and "Remove Mentor" based on the specified theme 
Implementation of a form for adding a new mentor, accessible via the "Add Mentor" link in the navigation panel.

2. Responsive Design:

Creation of both mobile and desktop versions of the application to ensure optimal user experience across different devices.
Utilization of design principles inspired by the "Book a Vespa Test-Ride" interface.
Integration of necessary inputs and elements to provide a seamless booking experience.

3. Redux Toolkit Integration:

Utilization of Redux Toolkit for state management to maintain consistency and scalability in managing application state.

4. Code Quality:

Adherence to best practices and coding standards to ensure maintainability and readability of the codebase.
Implementation of unit tests where applicable to validate the functionality of newly added components and features.
before deployment.

[Link to the back end repository PR](https://github.com/Ayokunnumi1/mentor_me_now_back_end/pull/10)

### CODE REVIEW

**Comment by lily4178993** [comment]
⚠️ In light of no feedback received from the reviewer, I'm moving forward with merging this pull request into the main branch and will now initiate the request for the final evaluation.

File: src/components/MentorDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Link, useParams } from 'react-router-dom';
+import { fetchMentorDetails } from '../redux/slices/mentors/mentorDetailsSlice';
+import LoadingStatus from './ux/LoadingStatus';
+import ErrorStatus from './ux/ErrorStatus';
+
+const MentorDetails = () => {
```
Comment (Meltrust) [inline src/components/MentorDetails.jsx:8]: - [ ] When  I go to a mentor's details and click on the "Reserve" button, it just takes me back to the homepage.

image


Kindly, make it so this button takes your users to the "Reserve Mentor" page, so we can meet this particular requirement. 👍 

_tip: Just make the reserve button link to the /reserveMentor route._

**Review by Meltrust — COMMENTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:


-  Nice code organization ✔️ 
-  Good readme ✔️ 
-  Nice frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/components/MentorDetails.jsx (hunk 1 above)
Comment (lily4178993) [reply src/components/MentorDetails.jsx:8]: Hello @Meltrust  Thanks for taking the time to review this PR.

I want to point out that this PR has already been merged into the main branch, and we've requested the final assessment. 

Could you please advise on where we should implement the necessary modifications - should it be in the develop branch or directly in main?

Thank you for your quick response, and I'll be eagerly awaiting further instructions.

**Comment by Meltrust** [comment]
Hey @lily4178993.  

In case you just want to add the changes then push to develop > merge to main.  

In case you want this review approved then push here to develop > review ok > merge to main.  Should be done in less than 5 minutes since it's re-review.

Never, ever, push directly to main.

**Pull request merged**